### PR TITLE
[Core][VTKOutput] make type of entity selectable

### PR DESCRIPTION
**Pull request by philbucher**
This PR makes the selection of entities written by the VtkOutput selectable, which was previously hardcoded
The behavior has not changed, still by default elements have a higher prio than conditions
Also the MPI-performance is slightly increased by removing some collective communication operations

This change is necessary for
- Preparation for #11140
- Preparation for Output of geometries

FYI @ncrescenzio

### CODE REVIEW

**Comment by sunethwarna** [comment]
@philbucher : I have a small question regarding this. Assume you want to write `Conditions` output and points output of a model part which has elements and conditions. In this case, we have to find out the nodes from `ModelPart::Nodes` which are relevant for conditions (and not consider the element nodes) ryt? Writing all the nodes and their data from `ModelPart::Nodes` will be wrong as far as I see. Am I missing something?

**Comment by philbucher** [comment]
This is true, all the nodes would be written

This would require extra "filtering".

Note though that this is also the existing behavior, not sure if we need the filtering. Could be added I guess if it becomes relevant

**Comment by philbucher** [comment]
Maybe we should also distinguish for nodes
all or from_entities

All is what we do now, but probably from_entities would be a better option for most cases

**Comment by philbucher** [comment]
I was waiting for your response/approval.

I believe there are still some discussions open in your PR, but we can merge this one already

**Comment by ncrescenzio** [comment]
Yes, but I think that the discussions in #11140 are not related to the topic of this PR. The selection of the entity type seems to work also there. So if other different changes are needed we can open a new PR, no?

**Comment by philbucher** [comment]
yes that makes sense

can you then please approve this PR?

**Review by ncrescenzio — APPROVED** [review]

**Pull request merged**
